azure-servicebus/azure/servicebus/aio/async_client.py

```diff
@@ -0,0 +1,654 @@
+        else:
+            entity_uri = self.mgmt_client.format_dead_letter_queue_name(self.entity_uri)
+        return Receiver(
+            self.loop,
+            handler_id,
+            entity_uri,
+            self.auth_config,
+            debug=self.debug,
+            timeout=int(idle_timeout * 1000),
+            prefetch=prefetch,
+            mode=mode,
+            **kwargs)
+
+
+class BaseClient(mixins.BaseClient):
+
+    def __init__(self, loop, address, name, shared_access_key_name=None,
+                 shared_access_key_value=None, debug=False, **kwargs):
+        """
+        Constructs a new Client to interact with the named ServiceBus entity.
+
+        :param loop: An async event loop.
+        :param address: The full URI of the Service Bus namespace. This can optionally
+         include URL-encoded access name and key.
+        :type address: str
+        :param name: The name of the entity to which the Client will connect.
+        :type name: str
+        :param shared_access_key_name: The name of the shared access policy. This must be supplied
+         if not encoded into the address.
+        :type shared_access_key_name: str
+        :param shared_access_key_value: The shared access key. This must be supplied if not encoded
+         into the address.
+        :type shared_access_key_value: str
+        :param debug: Whether to output network trace logs to the logger. Default is `False`.
+        :type debug: bool
+        """
+        self.loop = loop
+        super(BaseClient, self).__init__(
+            address, name, shared_access_key_name=shared_access_key_name,
+            shared_access_key_value=shared_access_key_value, debug=debug, **kwargs)
+
+    @classmethod
+    def from_entity(cls, loop, address, entity, **kwargs):  # pylint: disable=arguments-differ
+        client = cls(
+            loop,
+            address + "/" + entity.name,
+            entity.name,
+            validated_entity=entity,
+            **kwargs)
+        return client
+
+    @classmethod
+    def from_connection_string(cls, loop, conn_str, name=None, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Create a Client from a ServiceBus connection string.
+
+        :param loop: An async event loop.
+        :param conn_str: The connection string.
+        :type conn_str: str
+        :param name: The name of the entity, if the 'EntityName' property is
+         not included in the connection string.
+        """
+        address, policy, key, entity = parse_conn_str(conn_str)
+        entity = name or entity
+        address = build_uri(address, entity)
+        name = address.split('/')[-1]
+        return cls(loop, address, name, shared_access_key_name=policy, shared_access_key_value=key, **kwargs)
+
+    def _get_entity(self):
+        raise NotImplementedError("Must be implemented by child class.")
+
+
+class QueueClient(BaseClient, SendClientMixin, ReceiveClientMixin):
```

Comment:
Mixin should always be upfront
```python
class QueueClient(SendClientMixin, ReceiveClientMixin, BaseClient):
```
I don't have reference doc, but that's related to the MRO and some complicated usage of subclasses.